gaboon/commands/install.py

```diff
@@ -1,49 +1,301 @@
+            msg += (
+                "\n\nMissing or forbidden.\n"
+                "If this issue persists, generate a Github API token and store"
+                " it as the environment variable `GITHUB_TOKEN`:\n"
+                "https://github.blog/2013-05-16-personal-api-tokens/"
             )
+        raise ConnectionError(msg)
+
+    data = response.json()
+    if not data:
+        raise ValueError("Github repository has no tags set")
+    org, repo = data[0]["zipball_url"].split("/")[3:5]
+    tags = [i["name"].lstrip("v") for i in data]
+    if version not in tags:
+        raise ValueError(
+            "Invalid version for this package. Available versions are:\n"
+            + ", ".join(tags)
+        ) from None
+    return next(i["zipball_url"] for i in data if i["name"].lstrip("v") == version)
 
+
+def _pip_installs(
+    package_ids: list[str], base_install_path: Path, installer: str, quiet: bool = False
+):
+    logger.info(f"Installing {len(package_ids)} pip packages...")
+    cmd = []
     # TODO: Allow for multiple versions of the same package to be installed
-    cmd = ["uv", "pip", "install", *package_ids, "--target", str(path)]
+    if installer == "uv":
+        cmd = ["uv", "pip", "install", *package_ids, "--target", str(base_install_path)]
+    else:
+        cmd = ["pip", "install", *package_ids, "--target", str(base_install_path)]
 
     # TODO: `--upgrade` and `--force` options.
     capture_output = quiet
-    subprocess.run(cmd, capture_output=capture_output, check=True)
+    try:
+        subprocess.run(cmd, capture_output=capture_output, check=True)
+    except FileNotFoundError as e:
+        logger.info(
+            f"Stack trace:\n{''.join(traceback.format_exception(type(e), e, e.__traceback__))}"
+        )
+        logger.error(e)
+        logger.info(
+            f'Your installer {installer} is not found in your system PATH.\nPlease install {installer} or update your installer in your gaboon.toml as: \n\n[project]\ninstaller = "your installer here (pip or uv)"'
+        )
+        sys.exit(1)
+
+    _write_dependencies(package_ids, DependencyType.PIP)
+
 
+def _write_dependencies(new_package_ids: list[str], dependency_type: DependencyType):
     config = get_config()
     dependencies = config.get_dependencies()
+    typed_dependencies = [
+        dep for dep in dependencies if classify_dependency(dep) == dependency_type
+    ]
 
+    # Write to dependencies file
     to_delete = set()
-    for package in package_ids:
-        for dep in dependencies:
-            if Requirement(dep).name == Requirement(package).name:
-                to_delete.add(dep)
-        logger.info(f"Installed {package}")
+    if dependency_type == DependencyType.PIP:
+        for package in new_package_ids:
+            for dep in typed_dependencies:
+                if Requirement(dep).name == Requirement(package).name:
+                    to_delete.add(dep)
+            if package not in to_delete:
```

Comment:
i don't think this condition is right. it may have been updated.

Reply:
you were right, fixed in a1db36e